semgrep-core/CLI/Main.ml

```diff
@@ -745,23 +786,32 @@ let json_of_exn e =
 (*****************************************************************************)
 
 (*s: function [[Main_semgrep_core.semgrep_with_rules]] *)
-let semgrep_with_rules ~with_opt_cache rules files =
+let semgrep_with_rules ~with_opt_cache ~report_time rules files =
```

Comment:
Can we use globals for that ... in Flag_semgrep.ml
It's annoying to have to thread them. I like to use globals for those kinds of non-functional things.